unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/GLTF/AssetPromise_GLTF.cs

```diff
@@ -46,19 +46,24 @@ protected override void OnLoad(System.Action OnSuccess, System.Action OnFail)
                 useVisualFeedback = settings.visibleFlags == AssetPromiseSettings_Rendering.VisibleFlags.VISIBLE_WITH_TRANSITION,
                 initialVisibility = settings.visibleFlags != AssetPromiseSettings_Rendering.VisibleFlags.INVISIBLE,
                 shaderOverride = settings.shaderOverride,
-                addMaterialsToPersistentCaching = (settings.cachingFlags & MaterialCachingHelper.Mode.CACHE_MATERIALS) != 0
+                addMaterialsToPersistentCaching = (settings.cachingFlags & MaterialCachingHelper.Mode.CACHE_MATERIALS) != 0,
             };
 
-            tmpSettings.OnWebRequestStartEvent += ParseGLTFWebRequestedFile;
-
-            gltfComponent.LoadAsset(url, GetId() as string, false, tmpSettings);
+            gltfComponent.LoadAsset(provider.baseUrl ?? assetDirectoryPath, fileName, GetId() as string, false, tmpSettings, FileToHash);
             gltfComponent.OnSuccess += OnSuccess;
             gltfComponent.OnFail += OnFail;
 
-            asset.name = url;
+            asset.name = fileName;
         }
 
-        void ParseGLTFWebRequestedFile(ref string requestedFileName) { provider.TryGetContentsUrl(assetDirectoryPath + requestedFileName, out requestedFileName); }
+        bool FileToHash(string fileName, out string hash)
+        {
+            if (provider.TryGetContentHash(assetDirectoryPath + fileName, out hash))
+            {
+                return true;
+            }
+            return false;
```

Comment:
you could use
`return provider.TryGetContentHash(assetDirectoryPath + fileName, out hash);`

Reply:
Ups! I'll change it. 
I forget it like this while debugging because it easier to add breakpoints this way :P